packages/mui-material/src/StepButton/StepButton.test.js

```diff
@@ -145,4 +146,66 @@ describe('<StepButton />', () => {
 
     expect(getByRole('button')).not.to.equal(null);
   });
+
+  describe('prop: StepIconComponent', () => {
+    it('should render', () => {
+      function CustomizedIcon() {
+        return <div data-testid="custom-icon" />;
+      }
+      const { container, getByTestId } = render(
+        <Step active completed>
+          <StepButton StepIconComponent={CustomizedIcon}>Step One</StepButton>
+        </Step>,
+      );
+
+      const icon = container.querySelector(`.${stepLabelClasses.iconContainer}`);
+
+      getByTestId('custom-icon');
+      expect(icon).not.to.equal(null);
+      expect(icon).not.to.have.attribute('data-testid').equal('CheckCircleIcon');
+    });
+
+    it('should not render', () => {
+      const { container } = render(
+        <Step active completed>
+          <StepButton>Step One</StepButton>
+        </Step>,
+      );
+
+      const icon = container.querySelector(`.${iconClasses.root}`);
+      expect(icon).to.equal(null);
+    });
+
+    describe('renders <StepIcon> with the className completed', () => {
+      it('renders with completed className when completed', () => {
+        function CustomizedIcon() {
+          return <div data-testid="custom-icon" />;
+        }
+        const { container } = render(
+          <Step completed>
+            <StepButton StepIconComponent={CustomizedIcon}>Step One</StepButton>
+          </Step>,
+        );
+
+        const icon = container.querySelector(`.${stepLabelClasses.iconContainer}`);
+        expect(icon).to.have.class(stepLabelClasses.completed);
+      });
+    });
+
+    describe('renders <StepIcon> with the className active', () => {
+      it('renders with active className when active', () => {
+        function CustomizedIcon() {
+          return <div data-testid="custom-icon" />;
+        }
+        const { container } = render(
+          <Step active>
+            <StepButton StepIconComponent={CustomizedIcon}>Step One</StepButton>
+          </Step>,
+        );
+
+        const icon = container.querySelector(`.${stepLabelClasses.iconContainer}`);
+        expect(icon).to.have.class(stepLabelClasses.active);
+      });
+    });
```

Comment:
As mentioned previously [here](https://github.com/mui/material-ui/pull/39482#discussion_r1381839268), these tests are not needed. Please add tests for `StepIconProps` instead.